### PR TITLE
Reformatted code blocks for parents resuming responsibility for minor

### LINKED ISSUE #106: Out of order petitioner screens + error screen
This happened while running through the removal pathway as an interested party and only petitioner. Immediately after the initial temporary guardian screen (id: temporary guardian intro), the interview asked me to name two petitioners. After entering information for the second petitioner, the interview produced an error screen.

See attached PDF for "how questions came to be asked" section.
[Guardian Removal Error.pdf](https://github.com/user-attachments/files/17263907/Guardian.Removal.Error.pdf)

### PR DESCRIPTION
**Pull request by RamiLorca**
- Fixed #106
- Had to comment out `mandatory: True` in the main order block of the application_for_appointment_of_guardian.yml file, since it was interfering with the way the removal interview worked. 

While working to fix the issue, I found that it was connected to the gathering of responsible_parents (i.e. parents resuming responsibility for a minor). This led to including other fixes related to how responsible_parents are collected:

- The interview no longer tries to classify the minor's parents as mother/father. If a user indicates they are a parent, the interview will simply set their relationship to the minor as "Parent". 
- If a parent petitioner later indicates that the minor's are able to resume responsibility for the minor, the interview will then ask which parents should be included. 
- Depending on whether there's a second parent user, and whether that second parent is resuming responsibility as well, the interview will either automatically assign users as responsible_parents, or collect their information.
- Removed unused block dealing with wants_counsel_for_father and  wants_counsel_for_mother. The application for appointment of counsel is enabled by a block that checks whether the user is a parent and wants a lawyer.

### CODE REVIEW

File: docassemble/CLAGuardianship/data/questions/petition_for_removal_of_guardian.yml [hunk 1]
```diff
@@ -667,9 +822,13 @@ sets:
...
-  ${ capitalize(ordinal(i)) } parent resuming responsibility for
+  % if responsible_parents.target_number > 1:
+  ${ capitalize(ordinal(i)) } parent
+  % else:
+  Parent
+  % endif 
+  resuming responsibility for
   % if who_is_making_petition == "minor" and (not form_filled_by_attorney):
   you
   % else:
```
Comment (nonprofittechy) [inline docassemble/CLAGuardianship/data/questions/petition_for_removal_of_guardian.yml:841]: Avoid putting conditional logic into the middle of a sentence - it makes fragments that make it hard to translate into languages with different common word orders, like Spanish.

File: docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml [hunk 2]
```diff
@@ -162,7 +162,7 @@ code: |
 comment: |
   This block includes the logic for standalone interviews.
   Delete mandatory: True to include in another interview
-mandatory: True
+# mandatory: True
```
Comment (nonprofittechy) [inline docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml]: I don't think this is the right fix. ~Maybe instead, remove line 8 of the petition for removal? It shouldn't be including another interview that has its own launching point.~ (Just realized line 8 is a different interview)

**Review by nonprofittechy — CHANGES_REQUESTED** [review]
This change will break the petition for appointment. Can we figure out how the 2 forms are interacting now? I don't think they should be as the petition for appointment isn't directly referenced in the petition for removal.

File: docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml (hunk 2 above)
Comment (RamiLorca) [reply docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml]: @nonprofittechy I just pushed my latest changes that allow us to keep `mandatory: True` in both the appointment and removal files.

File: docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml [hunk 3]
```diff
@@ -166,6 +166,7 @@ mandatory: True
 code: |
   al_intro_screen
   petition_for_appointment_of_guardian_intro
+  set_guardian_removal
```
Comment (nonprofittechy) [inline docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml:169]:
```suggestion
```

File: docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml [hunk 4]
```diff
@@ -307,6 +308,10 @@ code: |
     
   set_no_assets = True
 ---
+code: |
+  guardian_removal = False
+  set_guardian_removal = True
```
Comment (nonprofittechy) [inline docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml:313]:
```suggestion
```

Comment (nonprofittechy) [reply docassemble/CLAGuardianship/data/questions/petition_for_appointment_of_guardian.yml:313]: You don't need a named block for this kind of variable setting

File: docassemble/CLAGuardianship/data/templates/petition_for_removal_of_guardian.pdf
Comment (nonprofittechy) [inline docassemble/CLAGuardianship/data/templates/petition_for_removal_of_guardian.pdf:1]: Just wanted to check to see if this was intentional and what the change was

**Review by nonprofittechy — APPROVED** [review]
I think there is a shorter way to set guardian_removal next time, but this works as-is

**Pull request merged**